src/hotspot/share/prims/jvmtiEventController.cpp

```diff
@@ -369,6 +369,7 @@ void JvmtiEventControllerPrivate::enter_interp_only_mode(JvmtiThreadState *state
   if (target == nullptr) { // an unmounted virtual thread
     return;  // EnterInterpOnlyModeClosure will be executed right after mount.
   }
+  ThreadsListHandle tlh(current);
```

Comment:
Why was this added?

Reply:
This is explained in the PR description.
Do you think, a just comment is needed or this has to be separated from this fix?

Reply:
I see the PR comment, but I don't really understand it. Is this to force some sort of early initialization to avoid a race later on? It just seems odd to create the tlh, but never use it.

Reply:
The `tlh` is used to protect any existing at this point JavaThread's from being terminated while it is set.
My understanding is that there is no need to iterate over all threads in the list to get this protection.

Reply:
Ok. I comment indicating that purpose would be useful.

Reply:
Okay, thanks!
I've added a comment:
```
  // Protects any existing JavaThread's from being terminated while it is set.
  // The FJP carrier thread compensating mechanism can create carrier threads concurrently.
```